web/views.py

```
@@ -5101,6 +5102,114 @@ def toggle_course_status(request, slug):
+            else:
+                messages.error(request, "Failed to mint NFT badge. Please try again later.")
+        except Exception as e:
+            messages.error(request, f"Error minting NFT badge: {str(e)}")
+
+        return render(request, "achievements/send_nft_badge.html", {"achievement": achievement})
+
+    return render(request, "achievements/send_nft_badge.html", {"achievement": achievement})
+
+
+@login_required
+def achievement_detail(request, achievement_id):
```

Ruff (0.8.2)

Missing return type annotation for public function `achievement_detail`

(ANN201)